modules/exploits/windows/smb/smb_relay.rb

```diff
@@ -69,678 +77,373 @@ module is not able to clean up after itself. The service and payload
-      'smb_pipeio' => 'rw',
-      'smb_client' => smb[:rclient]
-    }
-
-    handle = Rex::Proto::DCERPC::Handle.new([uuid, version], 'ncacn_np', smb[:ip], [pipe])
-    dcerpc = Rex::Proto::DCERPC::Client.new(handle, smb[:rsock], opts)
+  def smb_logger
+    log_device = datastore['VERBOSE'] ? RubySmbAdapter::Logging::LogDevice::Module.new(self) : RubySmbAdapter::Logging::LogDevice::Framework.new(framework)
+    RubySmbAdapter::Logging::Logger.new(self, log_device)
   end
 
-  def smb_cmd_dispatch(cmd, c, buff)
-    smb = @state[c]
+  class SMBRelayServer
+    include ::Rex::Proto
 
-    @pwned ||= {}
-
-    case cmd
-    when CONST::SMB_COM_NEGOTIATE
-      smb_cmd_negotiate(c, buff)
-
-    when CONST::SMB_COM_SESSION_SETUP_ANDX
-      smb_cmd_session_setup(c, buff)
-
-    when CONST::SMB_COM_TREE_CONNECT
-      print_status("Denying tree connect from #{smb[:name]}")
-      pkt = CONST::SMB_BASE_PKT.make_struct
-      pkt['Payload']['SMB'].v['Command'] = cmd
-      pkt['Payload']['SMB'].v['Flags1'] = 0x88
-      pkt['Payload']['SMB'].v['Flags2'] = 0xc001
-      pkt['Payload']['SMB'].v['ErrorClass'] = 0xc0000022
-      c.put(pkt.to_s)
-
-    else
-      print_status("Ignoring request from #{smb[:name]} (#{cmd})")
-      pkt = CONST::SMB_BASE_PKT.make_struct
-      pkt['Payload']['SMB'].v['Command'] = cmd
-      pkt['Payload']['SMB'].v['Flags1'] = 0x88
-      pkt['Payload']['SMB'].v['Flags2'] = 0xc001
-      pkt['Payload']['SMB'].v['ErrorClass'] = 0 # 0xc0000022
-      c.put(pkt.to_s)
+    def initialize(options)
+      @options = options
     end
-  end
-
-  def smb_cmd_negotiate(c, buff)
-    smb = @state[c]
-    pkt = CONST::SMB_NEG_PKT.make_struct
-    pkt.from_s(buff)
-
-    # Record the remote process ID
-    smb[:process_id] = pkt['Payload']['SMB'].v['ProcessID']
-
-    flags2 = pkt['Payload']['SMB'].v['Flags2']
-    extended_security = (flags2 & 0x800 == 0x800)
 
-    group = ''
-    machine = smb[:nbsrc]
-
-    dialects = pkt['Payload'].v['Payload'].gsub(/\x00/, '').split(/\x02/).grep(/^\w+/)
-    # print_status("Negotiation from #{smb[:name]}: #{dialects.join(", ")}")
-
-    dialect =
-      dialects.index('NT LM 0.12') ||
-      dialects.length - 1
-
-    # Dialect selected, now we try to the target system
-    target_host = datastore['SMBHOST']
-    if (!target_host || target_host.strip.empty?)
-      target_host = smb[:ip]
+    def alias
+      super || 'SMB Relay Server'
     end
 
-    # If extended security isn't supported or we're trying reflection
-    # ntlmv1 should be used, otherwise, use ntlmssp
-    if extended_security && target_host != smb[:ip]
-      smb[:ntlmssp] = true
-      negotiate_ntlmssp(smb, target_host)
-    else
-      smb[:ntlmssp] = false
-      negotiate_ntlmv1(smb, target_host)
+    #
+    # Returns the hardcore alias for the SMB service
+    #
+    def self.hardcore_alias(*args)
+      sock_options = sock_options_for(*args)
+      "#{sock_options['LocalHost']}#{sock_options['LocalPort']}"
```

Comment:
Related:

https://github.com/rapid7/metasploit-framework/issues/16314

Will align this with the approach taken by @smashery if a new convention is introduced as part of that bug fix